fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/AbstractDataAccessRule.java

```diff
@@ -457,10 +457,9 @@ private static Optional<ScanDirection> satisfiesRequestedOrdering(@Nonnull final
                     if (requestedSortOrder != OrderingPart.RequestedSortOrder.ANY) {
                         final var matchedSortOrder = orderingPart.getSortOrder();
 
-                        if ((matchedSortOrder == OrderingPart.MatchedSortOrder.ASCENDING &&
-                                     (requestedSortOrder == OrderingPart.RequestedSortOrder.ASCENDING)) ||
-                                (matchedSortOrder == OrderingPart.MatchedSortOrder.DESCENDING &&
-                                         (requestedSortOrder == OrderingPart.RequestedSortOrder.DESCENDING))) {
+                        // TODO: Will matchedSortOrder have counterflow nulls?
```

Comment:
Probably yes! So this looks like that if we have requested `DESCENDING` but with the incompatible `NULLS` we cannot satisfy the requested ordering in a way and return `Optional.empty()`. I think that is fine for now if that's the intention.

Reply:
Right. There isn't a case where the match works _except_ for null ordering; that's a mismatch.

But I wasn't sure the match sort order wouldn't be called on someplace to give additional information about its derivation.